core-backend/src/funcs.rs

```diff
@@ -286,31 +278,31 @@ where
 
     fn execute(
         self,
-        caller: &mut CallerWrap<Caller>,
+        caller: &mut MemoryCallerContext<Caller>,
         context: Self::Context,
         syscall_name: SyscallName,
     ) -> Result<(Gas, ()), HostError> {
         let Self { token, f, .. } = self;
-        let FallibleSyscallContext { gas, res_ptr } = context;
+        let FallibleSyscallContext { res_ptr, .. } = context;
```

Comment:
remove this trampoline for future errors from everywhere please
```suggestion
        let FallibleSyscallContext { res_ptr } = context;
```

Reply:
Ok